### PR TITLE
#170947542 setup docker

### PR DESCRIPTION
**Pull request by Ntare22**
**What does this PR do?**
Setup Docker

**Description of Task to be completed?**
* add Dockerfile
* add dockerignore file to ignore .dockerignore

**Any background context you want to provide?**
* N/A

**How should this be manually tested?**
* Install docker in your local machine
* run `docker build -t barefoot-nomad .`
* run `docker run -it -p 9000:3000 barefoot-nomad`
* go to your browser and check localhost:9000

**What are the relevant pivotal tracker stories?**
[#170947542](https://www.pivotaltracker.com/story/show/170947542)

**Screenshots (if appropriate)**
* N/A

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+npm-debug.log
```
Comment (Mnickii) [inline .dockerignore]: let's avoid this by configuring your editor

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM node:13
+
+WORKDIR /Users/jim.ntare/Documents/work/devrepublic-bn-backend
```
Comment (Mnickii) [inline Dockerfile]: I believe this  working directory is only specific to you. Change this to sth like ```/usr/src/index``` or something accessible by everyone

**Review by Baraka-Mugisha — DISMISSED** [review]

**Review by Mnickii — CHANGES_REQUESTED** [review]
In addition to the below review, we are missing a compose file to define our services

**Review by bbaime98 — DISMISSED** [review]

**Review by bbaime98 — DISMISSED** [review]

**Review by StevenDerrick — APPROVED** [review]

**Review by StevenDerrick — DISMISSED** [review]

**Comment by Mnickii** [comment]
Maybe add in the postgres image then this will be ready to go

**Review by Baraka-Mugisha — APPROVED** [review]

**Review by Mnickii — APPROVED** [review]

**Review by StevenDerrick — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+WORKDIR /home
+
+COPY package*.json /home/
+
+RUN npm install
+
+COPY . /home
+
+EXPOSE 3000
```
Comment (munyili-samuel) [inline Dockerfile:11]: This is great 👍 
If you deploy this image where port 3000 is in use, the API will not be accessible and in some cases, the image will not start.
A workaround here is to have
```
ENV PORT 3000
EXPOSE $PORT
```

When running this and you need to use a different port, you will just need to `export PORT=8000` or add it to the `docker-compose.yaml`

Comment (Ntare22) [reply Dockerfile:11]: Will adjust in my next PR..

**Pull request merged**